### PR TITLE
Support Node Image Version

### PR DESCRIPTION
**Pull request by gtxistxgao**
### Latest improvements:
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i><br>
### Contribution checklist:
- [x] I have reviewed the [documentation](https://github.com/Azure/adx-documentation-pr/wiki/Overall-basic-flow) for the workflow.
- [ ] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.
- [ ] The [OpenAPI Hub](https://aka.ms/openapiportal) was used for checking validation status and next steps.
### ARM API Review Checklist
- [ ] Service team MUST add the "WaitForARMFeedback" label if the management plane API changes fall into one of the below categories. 
- adding/removing APIs.
- adding/removing properties.
- adding/removing API-version. 
- adding a new service in Azure.

Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs.
- [ ] If you are blocked on ARM review and want to get the PR merged urgently, please get the ARM oncall for reviews (RP Manifest Approvers team under Azure Resource Manager service) from IcM and reach out to them. 
Please follow the link to find more details on [API review process](https://armwiki.azurewebsites.net/rp_onboarding/ResourceProviderOnboardingAPIRevieworkflow.html).

### CODE REVIEW

File: specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-04-01/managedClusters.json [hunk 1]
```diff
@@ -1478,6 +1478,10 @@
           "type": "string",
           "description": "Version of orchestrator specified when creating the managed cluster."
         },
+        "nodeImageVersion": {
```
Comment (yungezz) [inline specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-04-01/managedClusters.json]: is this a readonly property? adding new property is suggesting to be done in new API version.

Comment (yungezz) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-04-01/managedClusters.json]: https://github.com/Azure/adx-documentation-pr/wiki/Breaking-changes-guidelines

Comment (gtxistxgao) [reply specification/containerservice/resource-manager/Microsoft.ContainerService/stable/2020-04-01/managedClusters.json:1481]: this is not readonly. User can set it to "latest". This is a setting to be added to 2020-04-01, which should be a new API version right?

**Review by KrisBash — APPROVED** [review]
LGTM

**Comment by yungezz** [comment]
hi @gtxistxgao  is `2020-04-01` api version already public without this change? If true, this change is a breaking change, detected by CI tool also.

**Comment by gtxistxgao** [comment]
> hi @gtxistxgao is `2020-04-01` api version already public before this change? If true, this change is a breaking change, detected by CI tool also.

2020-04-01 has never publicly published. It only published behind a feature flag that only our internal subscription can use it.

**Comment by yungezz** [comment]
> > hi @gtxistxgao is `2020-04-01` api version already public before this change? If true, this change is a breaking change, detected by CI tool also.
> 
> 2020-04-01 has never publicly published. It only published behind a feature flag that only our internal subscription can use it.

thanks for confirm, I'll approve this with "ApprovedBreakingChange", and sdk owners, pls bump pay attention to version bumping when do releasing, this change contains sdk breaking change.

**Comment by gtxistxgao** [comment]
> > > hi @gtxistxgao is `2020-04-01` api version already public before this change? If true, this change is a breaking change, detected by CI tool also.
> > 
> > 
> > 2020-04-01 has never publicly published. It only published behind a feature flag that only our internal subscription can use it.
> 
> thanks for confirm, I'll approve this with "ApprovedBreakingChange", and sdk owners, pls bump pay attention to version bumping when do releasing, this change contains sdk breaking change.

I am curious why does this one have breaking change? It adds a new field in the 2020-04-01 API. Nothing removed from the previous API version. This PR also only merge to a dev branch. If check the master branch, 2020-04-01 API is not there for ContainerServer.

**Comment by yungezz** [comment]
@gtxistxgao  https://dev.azure.com/azure-sdk/public/_build/results?buildId=353715&view=logs&j=08b5038d-1e84-5877-88da-859df8312b69&t=c1978ecd-fc8a-5291-6d7a-7dd459d8b50f

- add property in response

**Comment by yungezz** [comment]
hi @NullMDR could you pls help to look at the SDK java error, it failed at pull docker image? https://dev.azure.com/azure-sdk/internal/_build/results?buildId=353713&view=logs&j=ffad7180-476d-5b8a-bb7f-738e424e0370&t=b8af71f8-27f2-53a9-4145-b7d64a49a6a5

**Comment by gtxistxgao** [comment]
> @gtxistxgao https://dev.azure.com/azure-sdk/public/_build/results?buildId=353715&view=logs&j=08b5038d-1e84-5877-88da-859df8312b69&t=c1978ecd-fc8a-5291-6d7a-7dd459d8b50f
> 
> * add property in response

This check makes sense if it is merge into master branch. But this PR merges into a dev branch. The assumption that "adding new property == breaking change" should not applied to this PR. Is this a new check? I did not see it last time.

**Pull request merged**